SalesforceCommerceCloud/pwa-kit
# W-14202811: Feature/product bundles

# Description
Implement products bundle into pwa-kit

Previously, there was a API limitation when basket could not retain bundle child variants. The issue has been resolve, and the child bundle variants are now present in the basket.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Checkout code
- Change the config default.js to use the sandbox zzrf-0014 to have product bundles
```
commerceAPI: {
            proxyPath: `/mobify/proxy/api`,
            parameters: {
                clientId: '3bffaaca-35af-4c2f-ad46-97f9ec01f7b2',
                organizationId: 'f_ecom_zzrf_014',
                shortCode: 'kv7kzm78',
                siteId: 'RefArchGlobal'
            }
        },
ssrParameters: {
        ssrFunctionNodeVersion: '16.x',
        proxyConfigs: [
            {
                host: 'kv7kzm78.api.commercecloud.salesforce.com',
                path: 'api'
            },
            {
                host: 'zzrf-014.dx.commercecloud.salesforce.com',
                path: 'ocapi'
            }
        ]
    }
```
- Run Retail App
- Go to this product http://localhost:3000/global/en-GB/product/test-bundle
- Choose variants for each bundle child
- Add bundle to cart
- Go to Cart page
- Check that all the bundle child variants are present in the basket


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
